spec/report_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+
+describe EasyPost::Report do
+  [:shipment, :tracker, :payment_log].each do |report_type|
+    context "report" do
+      let(:type) { report_type }
+      let(:type_factory) {
+        case type
+        when :shipment
+          "ShipmentReport"
+        when :tracker
+          "TrackerReport"
+        when :payment_log
+          "PaymentLogReport"
+        else
+          raise "Invalid type!"
+        end
+      }
+
+      describe '#create' do
+        it 'creates a report object' do
+          report = EasyPost::Report.create(
+            start_date: Date.today - 30,
+            end_date: Date.today,
+            type: type
+          )
+          expect(report.object).to eq type_factory
+          # will switch that test on when tracker status is deployed
+          # expect(report.status).to eq 'available'
+        end
+
+        it 'fails to create a report object' do
```

Comment:
This test gets run three times, even though its static. That's not bad at all (especially since it will simply error here, won't even make a request), but it does seem slightly odd to me. Can we pull it out of the each block?